src/fides/api/models/privacy_experience.py

```diff
@@ -327,23 +337,26 @@ def update_with_tcf_contents(
         if self.component == ComponentType.tcf_overlay:
             tcf_contents = copy(base_tcf_contents)
 
-            # Fetch previously saved records for the current user
-            for tcf_component, field_name in TCF_COMPONENT_MAPPING.items():
-                for record in getattr(tcf_contents, tcf_component):
+            has_tcf_contents = False
+            for (
+                tcf_section_name,
+                corresponding_db_field_name,
+            ) in TCF_SECTION_MAPPING.items():
+                # Loop through each top-level section in the TCF Contents
+                tcf_section: List = getattr(tcf_contents, tcf_section_name)
+                for record in tcf_section:
+                    # Loop through each record within a TCF section, and cache
+                    # any previously saved preferences if applicable
```

Comment:
I was just trying to reduce the amount of times I was looping through TCF_COMPONENT_MAPPING here.  This existing method takes the built TCF Experience Contents and adds each section to the requested Privacy Experience at runtime. If a fides user device id was supplied in the request, we also look to see if we have any saved preferences and attach them to each section.

Reply:
Ah! I didn't get a chance to experiment with this one to see if it helped here.  These specific queries should be performant but we know there are a lot of opportunities here.  I'll make a note to look into this further!